backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractTransitionEdgeDescriptionProvider.java

```diff
@@ -0,0 +1,139 @@
+import org.eclipse.sirius.components.view.builder.providers.IColorProvider;
+import org.eclipse.sirius.components.view.diagram.ArrowStyle;
+import org.eclipse.sirius.components.view.diagram.EdgeDescription;
+import org.eclipse.sirius.components.view.diagram.EdgeStyle;
+import org.eclipse.sirius.components.view.diagram.LabelEditTool;
+import org.eclipse.sirius.components.view.diagram.LineStyle;
+import org.eclipse.sirius.components.view.diagram.SynchronizationPolicy;
+import org.eclipse.syson.sysml.BindingConnectorAsUsage;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.util.AQLConstants;
+import org.eclipse.syson.util.AQLUtils;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+import org.eclipse.syson.util.ViewConstants;
+
+/**
+ * Used to create the {@link BindingConnectorAsUsage} edge description.
+ *
+ * @author adieumegard
+ */
+public abstract class AbstractTransitionEdgeDescriptionProvider extends AbstractEdgeDescriptionProvider {
+
+    public static final String NAME = "Edge TransitionUsage";
+
+    private final ViewBuilders viewBuilderHelper = new ViewBuilders();
+
+    private final DiagramBuilders diagramBuilderHelper = new DiagramBuilders();
+
+    private final IDescriptionNameGenerator descriptionNameGenerator;
+
+    public AbstractTransitionEdgeDescriptionProvider(IColorProvider colorProvider, IDescriptionNameGenerator descriptionNameGenerator) {
+        super(colorProvider);
+        this.descriptionNameGenerator = Objects.requireNonNull(descriptionNameGenerator);
+    }
+
+    @Override
+    public EdgeDescription create() {
+        String domainType = SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getTransitionUsage());
+        return this.getDiagramBuilderHelper().newEdgeDescription()
+                .domainType(domainType)
+                .isDomainBasedEdge(true)
+                .centerLabelExpression(AQLUtils.getSelfServiceCallExpression("getTransitionLabel"))
+                .name(this.getEdgeDescriptionName())
+                .semanticCandidatesExpression(AQLUtils.getSelfServiceCallExpression("getAllReachable", domainType))
+                .sourceNodesExpression(AQLConstants.AQL_SELF + ".source")
+                .style(this.createEdgeStyle())
+                .synchronizationPolicy(SynchronizationPolicy.SYNCHRONIZED)
+                .targetNodesExpression(AQLConstants.AQL_SELF + ".target")
+                .build();
+    }
+
+    protected String getEdgeDescriptionName() {
+        return this.getDescriptionNameGenerator().getDiagramPrefix() + NAME;
+    }
+
+    @Override
+    protected ChangeContextBuilder getSourceReconnectToolBody() {
+        return this.getViewBuilderHelper().newChangeContext()
+                .expression(AQLConstants.AQL + AQLConstants.EDGE_SEMANTIC_ELEMENT + ".reconnectSourceTransitionEdge(" + AQLConstants.SEMANTIC_RECONNECTION_TARGET + ")");
```

Comment:
You should be able to use on of the method from `AQLUtils` instead. Please try to change that.